Formula/apache-arrow.rb

```diff
@@ -40,6 +37,8 @@ def install
       -DARROW_USE_GLOG=OFF
       -DARROW_PYTHON=OFF
       -DARROW_S3=ON
+      -DARROW_WITH_BROTLI=ON
```

Comment:
Comparing to https://github.com/apache/arrow/blob/master/dev/tasks/homebrew-formulae/autobrew/apache-arrow.rb, we also should have this on (though up above, sorted alphabetically; I just can't suggest up there)

```suggestion
      -DARROW_GCS=ON
      -DARROW_WITH_BROTLI=ON
```

Reply:
(Likewise for the other variations on this formula)

Reply:
What is gcs? Does this not need any extra deps?

Reply:
google cloud storage. google-cloud-cpp and its dependencies (notably abseil-cpp) get built in the Arrow C++ build if they're not found, as with any other dependencies. We could certainly set up an autobrew formula for it, or for abseil, but it's not strictly required, it just makes the arrow build take a few extra minutes.